dashboard/src/main/java/com/google/cloud/tools/opensource/dashboard/DashboardMain.java

```diff
@@ -391,45 +392,31 @@ private static ArtifactResults generateArtifactReport(
   }
 
   /**
-   * Returns a table where rows are jar files in {@code jarFilter}, columns are {@link
-   * SymbolProblem}, and values are class names referencing the problems. When {@code jarFilter} is
-   * {@code null}, jar files are not filtered.
+   * Returns a nested map where first keys are JAR files with symbol problems, second keys are the
+   * problems, and values are class names in those JAR files with the problems.
    *
-   * <p>For example, {@code classes = table.get(JarX, SymbolProblemY)} means that {@code JarX} has
-   * {@code SymbolProblemY} and that {@code classes} in {@code JarX} reference the symbol of {@code
-   * SymbolProblemY}.
+   * <p>For example, {@code classes = table.get(JarX).get(SymbolProblemY)} where {@code classes} are
+   * not null means that {@code JarX} has {@code SymbolProblemY} and that {@code JarX} contains
+   * {@code classes} which reference {@code SymbolProblemY.getSymbol()}.
    */
-  private static ImmutableTable<Path, SymbolProblem, ImmutableSet<String>> indexByJar(
-      ImmutableSetMultimap<SymbolProblem, ClassFile> symbolProblems,
-      @Nullable Set<Path> jarFilter) {
-    Table<Path, SymbolProblem, ImmutableSet.Builder<String>> table = HashBasedTable.create();
-
-    symbolProblems.asMap().forEach((symbolProblem, classFiles) -> {
-      ImmutableListMultimap<Path, ClassFile> jarToClassFiles =
-          Multimaps.index(classFiles, ClassFile::getJar);
-      jarToClassFiles.forEach((jar, classFilesWithJar) -> {
-        if (jarFilter != null && !jarFilter.contains(jar)) {
-          return;
-        }
-        ImmutableSet.Builder<String> classNames = table.get(jar, symbolProblem);
-        if (classNames == null) {
-          classNames = ImmutableSet.builder();
-          table.put(jar, symbolProblem, classNames);
-        }
-
-        ImmutableSet<String> newClassNames = classFiles.stream().map(ClassFile::getClassName)
-            .collect(toImmutableSet());
-        classNames.addAll(newClassNames);
-      });
-    });
-
-    return table.cellSet()
-        .stream()
-        .collect(toImmutableTable(
-            Cell::getRowKey,
-            Cell::getColumnKey,
-            cell -> cell.getValue().build()
-        ));
+  private static ImmutableMap<Path, ImmutableSetMultimap<SymbolProblem, String>> indexByJar(
+      ImmutableSetMultimap<SymbolProblem, ClassFile> symbolProblems) {
+
+    ImmutableMap<Path, Collection<Entry<SymbolProblem, ClassFile>>> jarMap = Multimaps
+        .index(
+            symbolProblems.entries(),
+            entry -> entry.getValue().getJar()).asMap();
```

Comment:
Please run google-java-format. It'll make this expression easier to read because it'll be clear that `asMap()` is being called on the result of `Multimaps.index()`.